dotCMS/src/main/java/com/dotcms/languagevariable/business/LanguageVariableAPIImpl.java

```diff
@@ -120,4 +140,72 @@ private String getValueFromDefaultLanguage(final String key, final User user, fi
     return (null != keyValue) ? keyValue.getValue() : null;
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Map<Language, List<LanguageVariable>> findAllVariables() {
+    final List<Language> languages = languageAPI.getLanguages();
+
+    return languages.stream()
+            .collect(Collectors.toMap(lang -> lang, lang -> {
+              try {
+                return findVariables(lang.getId());
+              } catch (DotDataException e) {
+                Logger.error(this, "Error finding language variables", e);
+                return List.of();
+              }
+            }));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @CloseDBIfOpened
+  @Override
+  public List<LanguageVariable> findVariables(final long langId) throws DotDataException {
+
+    final ContentType contentType = langVarContentType.get();
+    final LanguageCache languageCache = CacheLocator.getLanguageCache();
+    final List<LanguageVariable> cacheVars = languageCache.getVars(langId);
+    if (cacheVars != null && !cacheVars.isEmpty()) {
```

Comment:
Empty should be a valid state in the cache, meaning, I guess you could have a language without variables and in that case, if the empty is already in cache we don't need to query again the DB.
I think that validation should check only for `!= null`.

Reply:
Yes, I can use an empty marker here. The 404 cache Pattern

Reply:
check out the new method I added languageCache#ifPresentGetOrElseFetch
it is meant to solve this situation and use empty List if returned by the data access layer 
without making cache return null